### PR TITLE
HomeWork 14

### PR DESCRIPTION
**Pull request by verovkin**
working on third one… the toughest one…

### CODE REVIEW

File: 14/phonebook.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    else:
+        print("empty")
+
+
+def show_from_phonebook(name):
+    print(f"{name} {phone_book[name] if name in phone_book else 'not found'}")
+
+
+def validate_phone(phone):
+    regex = "(\+?38)?0\d{9}$"
```
Comment (verovkin) [inline 14/phonebook.py:81]: validation function

File: 14/email_change2.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+# check if file exist:
+if not path.exists(txt_file):
+    print(f"File '{txt_file}' not found")
+    exit()
+
+with open(txt_file, 'r') as f:
+    text = f.read()
+
+email_regex = re.compile(r'(?<=.).(?=[^@]*?@)|(?:(?<=@)|(?!^)(?=[^@]*$))(.)(?=.*\.)|(?:(?<=\.)(?=[^.]+$)|(?!^)(?=[^@.]*$))[^.](?!$)', re.MULTILINE)
```
Comment (klymenok) [inline 14/email_change2.py]: Even don't know how to debug this, but in your case, the replacement keeps only the last letter of an email. And it should keep both first and last.

**Review by klymenok — CHANGES_REQUESTED** [review]

**Comment by verovkin** [comment]
Edits done.
made a bit easier that RegEx

**Review by klymenok — APPROVED** [review]
Looks good!

**Pull request merged**